src/Corax/Queries/BinaryMatch.Erasure.cs

```diff
@@ -85,7 +96,17 @@ static int AndWithFunc(ref BinaryMatch match, Span<long> matches)
                     return 0;
                 }
 
-                FunctionTable = new FunctionTable(&FillFunc, &AndWithFunc, &CountFunc);
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                static void ScoreFunc(ref BinaryMatch match, Span<long> matches, Span<float> scores)
+                {
+                    if (match._inner is BinaryMatch<TInner, TOuter> inner)
+                    {
+                        inner.Score(matches, scores);
+                        match._inner = inner;
```

Comment:
Why do you set it to the same value that it is already set to? Because the `Score()` call may have modify the struct? 
Aren't we doing the call in place? Or did we have boxing / unboxing here?

Reply:
There is a transitory boxing and unboxing happening here (technically) but the JIT somehow is just using a copy of the struct, that is why it get set afterwards.